### ISSUE TITLE
Doesn't catch 'ended' events from videojs on OSX

### ISSUE THREAD

**Issue by redbar0n** [thoughtbot/capybara-webkit#733]
I have some feature tests that include playback of an MP4 video file, using capybara-webkit and video.js. With Qt5 installed, the file plays and everything.

But capybara-webkit doesn't seem to catch ['ended'](https://github.com/videojs/video.js/blob/stable/docs/api/vjs.Player.md#ended-event) events fired by video.js, even though Chrome, Firefox and Selenium does.

from [video.js](https://github.com/videojs/video.js):

```
/**
 * Fired when the end of the media resource is reached (currentTime == duration)
 * @event ended
 */
vjs.Player.prototype.onEnded = function(){
  console.log("VIDEOJS END OF VIDEO");
  this.addClass('vjs-ended');
  if (this.options_['loop']) {
    this.currentTime(0);
    this.play();
  } else if (!this.paused()) {
    this.pause();
  }
};
```

Here is the failing test case and webkit_debug output:
https://gist.github.com/redbar0n/1a2a201418e626357dfa

My system specs:
OSX Mavericks 10.9.5
capybara 2.4.4
capybara-webkit 1.4.1
Qt 5.4.1
video.js 4.12.1
jQuery 1.10.0

**Comment by redbar0n** [comment]
The funny thing is that onPlay events seem to be caught.

From the video.js source:

```
/**
 * Fired whenever the media begins or resumes playback
 * @event play
 */
vjs.Player.prototype.onPlay = function(){
  console.log("ONPLAY VIDEOJS EVENT CATCHED");
  this.removeClass('vjs-ended');
  this.removeClass('vjs-paused');
  this.addClass('vjs-playing');

  // hide the poster when the user hits play
  // https://html.spec.whatwg.org/multipage/embedded-content.html#dom-media-play
  this.hasStarted(true);
};
```

**Comment by redbar0n** [comment]
The slider seems to indicate the video is played to the end:
```
Finished "Evaluate($(".vjs-progress-holder.vjs-slider").attr("aria-valuenow");)" with response "Success("100")"
Wrote response true ""100""
```

But it seems like the video isn't actually played to the end so the 'ended' event is fired and the .onEnded listener can respond to it.

This might be related to a previous issue in Chrome, but which Chrome has fixed:
https://code.google.com/p/chromium/issues/detail?id=157543

**Comment by redbar0n** [comment]
I was able to circumvent this problem by manually listening for video time changes, and firing the 'ended' event just before the video is complete (and thus preventing the normal 'ended' event from firing, so that there will only be one 'ended' event firing even in browsers which would otherwise fire the normal one):


	var notEnded = true; // global variable

	unitPlayer.on('timeupdate', function(e){
		var duration = unitPlayer.duration();
		if(notEnded && duration>0 && (duration-unitPlayer.currentTime()<0.15)) {
			unitPlayer.trigger('ended');
			notEnded = false; // must be there to prevent 'timeupdate' from triggering the 'ended' event twice, since 'timeupdate' executes every 15-250 milliseconds, depending on the playback technology in use.
		}
	});

**Comment by calleerlandsson** [comment]
@redbar0n Are you having these issues using the latest version of capybara-webkit? If so, could you extract a minimal failing example? This would help a lot in debugging this.

**Comment by redbar0n** [comment]
Currently using `capybara-webkit 1.5.2`, and all other system specifications the same as before.

With my workaround (in my previous comment) it still works and all tests run green.
When I comment out my workaround, then the tests run/stall forever, similar to before, but with a slightly different output.

Here's a gist of what I'm currently seeing:
https://gist.github.com/redbar0n/82d253e319c7e120bcdb

Not sure how I would go about extracting a minimal failing example in addition to that.

**Comment by redbar0n** [comment]
Now I upgraded to capybara-webkit `1.6.0`, and the results are exactly the same as with `1.5.2`. Still failing.

**Comment by twalpole** [comment]
Closing due to age and the lack of a minimal failing example.  If this is still an issue we can reopen when a way to replicate is provided

**Comment by redbar0n** [comment]
Did you see the gists I provided?

> Here is the failing test case and webkit_debug output:
> https://gist.github.com/redbar0n/1a2a201418e626357dfa
> Currently using capybara-webkit 1.5.2, and all other system specifications the same as before:
> https://gist.github.com/redbar0n/82d253e319c7e120bcdb

**Comment by twalpole** [comment]
@redbar0n capybara-webkit 1.5.2 was released over 2 years ago -- current release is 1.14.0 -- and the gist doesn't provide a minimal example we can use to replicate the issue.

**Comment by redbar0n** [comment]
Ok.
